From 4099d491a604ca9d77ae761bcd697b52ac7160b4 Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Wed, 24 Jul 2013 14:49:22 +0200
Subject: [PATCH 23/44] NFC: Fix SE discovery failure warning condition

This is a typo coming from the initial implementation. se_discover fails
when it returns something different than zero and we should only display
a warning in that case.

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
(cherry picked from commit 21d9c5637088b07cea83267cbd1845b773274493)
---
 net/nfc/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/nfc/core.c b/net/nfc/core.c
index 1d074dd..aad7f8f 100644
--- a/net/nfc/core.c
+++ b/net/nfc/core.c
@@ -129,7 +129,7 @@ int nfc_dev_up(struct nfc_dev *dev)
 	/* We have to enable the device before discovering SEs */
 	if (dev->ops->discover_se) {
 		rc = dev->ops->discover_se(dev);
-		if (!rc)
+		if (rc)
 			pr_warn("SE discovery failed\n");
 	}
 
-- 
1.8.4.93.g57e4c17

